cli/pkg/kctrl/cmd/app/init/init.go

```diff
@@ -0,0 +1,336 @@
+
+	//As fetchSource is not from Local directory, we should add upstream folder in ytt path and helmTemplate(if required).
+	isHelmTemplateRequired := false
+	if fetchSource == fetch.FetchChartFromHelmRepo || fetchSource == fetch.FetchChartFromGithub {
+		isHelmTemplateRequired = true
+	}
+	if isHelmTemplateRequired {
+		//TODO Figure out the edge scenarios
+		addUpstreamAsPathToHelmIfNotExist(templateSectionFromExistingApp)
+		addStdInAsPathToYttIfNotExist(templateSectionFromExistingApp)
+
+	}
+
+	addUpstreamAsPathToYttIfNotExist(templateSectionFromExistingApp)
+
+	return existingApp, nil
+}
+
+func (createStep CreateStep) configureExportSection() {
+	fetchSource := createStep.appBuild.ObjectMeta.Annotations[fetch.FetchContentAnnotationKey]
+	if fetchSource == fetch.FetchFromLocalDirectory {
+		return
+	}
+
+	// TODO current implementation is if export section is already defined, we will not touch it. Confirm the same.
+	if createStep.appBuild.Spec.Export == nil || len(createStep.appBuild.Spec.Export) == 0 {
+		createStep.appBuild.Spec.Export = append(createStep.appBuild.Spec.Export, appbuild.Export{
+			ImgpkgBundle: nil,
+			IncludePaths: []string{UpstreamFolderName},
+		})
+	}
+	return
+}
+
+func addUpstreamAsPathToYttIfNotExist(appTemplates []kcv1alpha1.AppTemplate) {
+	for _, appTemplate := range appTemplates {
+		if appTemplate.Ytt != nil {
+			for _, path := range appTemplate.Ytt.Paths {
+				if strings.HasPrefix(path, UpstreamFolderName) {
+					return
+				}
+			}
+		}
+	}
+	appTemplateWithYtt := kcv1alpha1.AppTemplate{
+		Ytt: &kcv1alpha1.AppTemplateYtt{
+			Paths: []string{UpstreamFolderName},
+		},
+	}
+	appTemplates = append([]kcv1alpha1.AppTemplate{appTemplateWithYtt}, appTemplates...)
+}
+
+func addStdInAsPathToYttIfNotExist(appTemplates []kcv1alpha1.AppTemplate) {
+	for _, appTemplate := range appTemplates {
+		if appTemplate.Ytt != nil {
+			for _, path := range appTemplate.Ytt.Paths {
+				if path == StdIn {
+					return
+				}
+			}
+		}
+	}
+	appTemplateWithYtt := kcv1alpha1.AppTemplate{
+		Ytt: &kcv1alpha1.AppTemplateYtt{
+			Paths: []string{StdIn},
+		},
+	}
+
+	//YttTemplate with Stdin should be the immediate next template to the helmTemplate.
+	index := 0
+	var appTemplate kcv1alpha1.AppTemplate
+	for index, appTemplate = range appTemplates {
+		if appTemplate.HelmTemplate != nil {
+			continue
+		}
+	}
+	var oldAppTemplates []kcv1alpha1.AppTemplate
+	copy(oldAppTemplates, appTemplates)
+	appTemplates = append(oldAppTemplates[:index], appTemplateWithYtt)
+	appTemplates = append(appTemplates, oldAppTemplates[index:]...)
+
+	return
+}
+
+func addUpstreamAsPathToHelmIfNotExist(appTemplates []kcv1alpha1.AppTemplate) {
+	for _, appTemplate := range appTemplates {
+		if appTemplate.HelmTemplate != nil {
+			path := appTemplate.HelmTemplate.Path
+			//If a helmTemplate exist, it will always be the first one in the template section. Theoretically, it can exist anywhere but every real use case needs it to be first.
```

Comment:
If a helmTemplate exist, it will always be the first one in the template section. Theoretically, it can exist anywhere but every real use case needs it to be first. Not handled the scenario if helmTemplate exists as not the first element of AppTemplate slice.

Reply:
well... sometimes there might be sops, for example as a first step.

Reply:
If appTemplate section is defined already, we will not touch it and return immediately.